src/producer/Producer.ts

```
@@ -121,28 +129,31 @@ export default class Producer {
     records: BaseAvroRecord[],
     topic: string,
     produceFailureRecords: boolean
-  ): Promise<Buffer[]> =>
-    await Promise.all(
-      records.map(async record => {
-        try {
-          return await this.serializer.encodeMessage(topic, record.schema(), record);
-        } catch (err) {
-          this.handleEncodingError(err, record, topic, produceFailureRecords);
-        }
-      })
-    );
```

Found this issue from running integration/Producer.test.ts and seeing that the error was thrown outside of the promise chain (the test would die and the error was not handled by jest). I think it had something to do with unwrapping the promise in this method. With `Promise.all` I think you just need to makes sure you are giving it an array of promises; it's hard to tell but I think the internal awaiting/try/catching was making it so that it was getting something like [Promise, Promise, Error, Promise] which causes the error to not be handled in the promise chain.

---

Let me know if you want some assistance with a test.